pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -294,8 +294,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	doBinaryCopyInit := true // copy rook binary in an init container?
 
 	// This property is used for both PVC and non-PVC use case
-	if osd.CVMode == "" {
-		return nil, errors.Errorf("failed to generate deployment for OSD %d. required CVMode is not specified for this OSD", osd.ID)
+	if osd.CVMode != "raw" && osd.CVMode != "lvm" {
```

Comment:
Are there any other possible CVModes than raw or lvm? If not, then I see this is equivalent to the previous condition.

Reply:
No. CVMode is always "raw" or "lvm". Reverted this change, thanks.

Reply:
That's ok if you want that condition to be clear about which types are available, I just wanted to be clear.